hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -464,4 +487,138 @@ private static Account getEffectiveOwnerAccount(
+     * @return the number of new crypto allowances that will be added to state from the transaction
+     */
+    private int getChangedCryptoKeys(
+            final List<CryptoAllowance> newAllowances, final List<AccountCryptoAllowance> existingAllowances) {
+        int counter = 0;
+        for (var key : newAllowances) {
+            for (int i = 0; i < existingAllowances.size(); i++) {
+                if (existingAllowances.get(i).spenderId().equals(key.spender())) {
+                    break;
+                }
+                counter++;
+            }
+        }
+        return counter;
+    }
+
+    /**
+     * Gets the number of new token allowances that will be added to state from the transaction.
+     * @param newAllowances the list of new token allowances
+     * @param existingAllowances the list of existing token allowances
+     * @return the number of new token allowances that will be added to state from the transaction
+     */
+    private int getChangedTokenKeys(
+            final List<TokenAllowance> newAllowances, final List<AccountFungibleTokenAllowance> existingAllowances) {
+        int counter = 0;
+        for (final var key : newAllowances) {
+            for (final var existingAllowance : existingAllowances) {
+                if (existingAllowance.spenderId().equals(key.spender())
+                        && existingAllowance.tokenId().equals(key.tokenId())) {
+                    break;
+                }
+                counter++;
+            }
+        }
+        return counter;
+    }
+
+    /**
+     * Gets the number of new approveForAllNftAllowances that will be added to state from the transaction.
+     * @param newAllowances the list of new nft allowances
+     * @param existingAllowances the list of existing nft allowances
+     * @return the number of new approveForAllNftAllowances that will be added to state from the transaction
+     */
+    private int getChangedNftKeys(
+            final List<NftAllowance> newAllowances, final List<AccountApprovalForAllAllowance> existingAllowances) {
+        int counter = 0;
+        for (final var key : newAllowances) {
+            for (final var existingAllowance : existingAllowances) {
+                if (existingAllowance.spenderId().equals(key.spender())
+                        && existingAllowance.tokenId().equals(key.tokenId())) {
+                    break;
+                }
+                counter++;
+            }
+        }
+        return counter;
+    }
+
+    /**
+     * Counts the number of serials in the list of nft allowances. It doesn't consider duplicates.
+     * @param nftAllowancesList the list of nft allowances
+     * @return the number of serials
+     */
+    private int countSerials(final List<NftAllowance> nftAllowancesList) {
+        int totalSerials = 0;
+        for (var allowance : nftAllowancesList) {
+            totalSerials += allowance.serialNumbers().size();
+        }
+        return totalSerials;
```

Comment:
Could instead use
`return nftAllowancesList.stream().mapToInt(nftAllowance -> nftAllowance.serialNumbers().size()).sum();`

Reply:
yeah , did not use streams intentionally. I am not sure if now we are allowed to use. Previously, we never used streams in mono-service since they are expensive.